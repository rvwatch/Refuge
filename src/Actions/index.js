export const addUser = user => ({
  type: 'ADD_USER',
  user
});
