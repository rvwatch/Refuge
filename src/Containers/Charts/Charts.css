
.key-container {
  padding: 5px;
  position: relative;
  font: 20px;
}

.resting {
  background-color: green;
  font-size: 16px;
  margin-left: 30px;
  padding: 3px 10px;
  color: #fff;
  border-radius: 5px;
}

.current {
  background-color: #83B8F4;
  font-size: 16px;
  color: #000;
  margin-left: 20px;
  padding: 3px 10px;
  border-radius: 5px;
}

.steps {
  background-color: #ffcf32;
  font-size: 16px;
  color: #000;
  margin-left: 20px;
  padding: 3px 10px;
  border-radius: 5px;
}

.chart-wrap {
  display: block;
  position: relative;
  margin: 30px;
  height: 450px;
  overflow-x: scroll;
}



.chart-key {
  position: absolute;
  top: 20px;
  left: 30px;
}

.chart-wrap::-webkit-scrollbar {
  -webkit-appearance: none;
  width: 7px;
}
.chart-wrap::-webkit-scrollbar-thumb {
  border-radius: 4px;
  background-color: rgba(0,0,0,.5);
  box-shadow: 0 0 1px rgba(255,255,255,.5);
}

.chart-wrap section{
  background-color: #fff;
  width: 100%;
  height: 300px;
  margin: 10px 0 0 0;
  border-radius: 10px;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
}

.VictoryContainer {
  background-color: #fff;
  height: 90% !important; /* yes... I'm aware. This stays until I figure out Victory.js styles */
  width: 2000px !important;
  border-radius: 10px;
  -webkit-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  -moz-box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,0.35);
  margin-top: 15px;
}