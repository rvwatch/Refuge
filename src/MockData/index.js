export const mockHeartRate = [
  { time: '00:00:00', value: 57 },
  { time: '00:01:00', value: 59 },
  { time: '00:02:00', value: 58 },
  { time: '00:03:00', value: 59 },
  { time: '00:04:00', value: 59 },
  { time: '00:05:00', value: 59 }
];

export const mockChartData = [
  { x: '12:00:00 A.M.', y: 57 },
  { x: '12:01:00 A.M.', y: 59 },
  { x: '12:02:00 A.M.', y: 58 },
  { x: '12:03:00 A.M.', y: 59 },
  { x: '12:04:00 A.M.', y: 59 },
  { x: '12:05:00 A.M.', y: 59 }
];

export const mockStepsTaken = [
  { time: '00:00:00', value: 0 },
  { time: '00:01:00', value: 0 },
  { time: '00:02:00', value: 0 },
  { time: '00:03:00', value: 0 },
  { time: '00:04:00', value: 0 },
  { time: '00:05:00', value: 0 },
  { time: '00:06:00', value: 0 },
  { time: '00:07:00', value: 0 }
];

export const mockFitbitProfile = {
  user: 'Ricardo V.',
  avgSteps: 8150
};

export const restingHeartRate = '72.34';

export const user = { username: 'ricardo' };
