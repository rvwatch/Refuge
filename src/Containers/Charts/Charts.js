import React, { Component } from 'react';
import { connect } from 'react-redux';
import { string, array } from 'prop-types';
import { Marker } from '../../Components/Marker/Marker';
import {
  VictoryChart,
  VictoryLine,
  VictoryAxis,
  VictoryBar,
  VictoryLabel
} from 'victory';
import './Charts.css';

export class Charts extends Component {
  constructor() {
    super();
    this.state = {
      markers: []
    };
  }

  render() {
    const { heartRate, stepsTaken, restingHeart } = this.props;
    const chartData = dataStream => {
      return dataStream.map(point => {
        const time = point.time.split(':');
        const hours = Number(time[0]);
        const minutes = Number(time[1]);
        const seconds = Number(time[2]);
        let timeValue;
        if (hours > 0 && hours <= 12) {
          timeValue = '' + hours;
        } else if (hours > 12) {
          timeValue = '' + (hours - 12);
        } else if (hours === 0) {
          timeValue = '12';
        }

        timeValue += minutes < 10 ? ':0' + minutes : ':' + minutes; // get minutes
        timeValue += seconds < 10 ? ':0' + seconds : ':' + seconds; // get seconds
        timeValue += hours >= 12 ? ' P.M.' : ' A.M.'; // get AM/PM
        const newTime = {
          // eslint-disable-next-line
          x: timeValue,
          // eslint-disable-next-line
          y: point.value
        };

        return newTime;
      });
    };

    const renderMarkers = this.state.markers.map(mark => {
      return <Marker key={mark.yPos} {...mark} />;
    });

    const showCoords = event => {
      const container = document.querySelector('.chart-wrap');
      const xPos = event.clientX - container.offsetLeft - 10 + container.scrollLeft;
      const yPos = event.clientY - 125;
      const markers = [...this.state.markers, {xPos, yPos}];
      this.setState({ markers });
    };

    const heartLine = chartData(heartRate);
    const stepLine = chartData(stepsTaken);
    const restingLine = chartData(
      heartRate.map(beat => ({
        time: beat.time,
        value: parseInt(restingHeart, 10)
      }))
    );

    return (
      <div className="key-container">
        <h2 className="chart-key">
          daily activity:
          <span className="resting">resting bpm--</span>
          <span className="current">latest bpm--</span>
          <span className="steps">steps taken--</span>
        </h2>

        <section className="chart-wrap" onClick={showCoords}>
          {renderMarkers}
          <VictoryChart height={375} width={2000}>
            <VictoryLine
              interpolation="monotoneX"
              data={heartLine}
              style={{ data: { stroke: '#83B8F4' } }}
            />
            <VictoryLine
              interpolation="monotoneX"
              data={restingLine}
              style={{ data: { stroke: 'green' } }}
            />
            <VictoryBar
              barRatio={2}
              style={{ data: { fill: '#ffcf32' } }}
              data={stepLine}
            />

            <VictoryAxis
              // eslint-disable-next-line
              tickFormat={t => {
                const num = t.split(':');
                if (num[1] === '00') {
                  return num[0];
                } else {
                  return '';
                }
              }}
            />

            <VictoryAxis dependentAxis orientation="left" domain={[0, 220]} />
            <VictoryAxis dependentAxis orientation="right" />
          </VictoryChart>
        </section>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  heartRate: state.heartRate,
  stepsTaken: state.stepsTaken,
  restingHeart: state.restingHeart
});

Charts.propTypes = {
  heartRate: array,
  stepsTaken: array,
  restingHeart: string
};

export default connect(mapStateToProps, null)(Charts);
