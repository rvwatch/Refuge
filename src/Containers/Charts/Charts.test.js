import React from 'react';
import { shallow } from 'enzyme';
import { Charts, mapStateToProps, chartData} from './Charts';
import * as Actions from '../../Actions/';
import * as mock from '../../MockData/';

describe('Charts', () => {
  let wrapper;
  let heartRate;
  let stepsTaken;
  beforeEach(() => {
    stepsTaken = mock.mockStepsTaken;
    heartRate = mock.mockHeartRate;
    wrapper = shallow(<Charts heartRate={heartRate} stepsTaken={stepsTaken} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should clean heartRate data', () => {
    const results = chartData(mock.mockHeartRate);
    const expected = mock.mockChartData;
    expect(results).toEqual(expected);
  });

  describe('mapStateToProps', () => {
    it('should add heartRate data to props', () => {
      const expected = mock.mockHeartRate;
      const mockState = { heartRate: mock.mockHeartRate };
      const mapped = mapStateToProps(mockState);
      expect(mapped.heartRate).toEqual(expected);
    });

    it('should add stepsTaken data to props', () => {
      const expected = mock.mockStepsTaken;
      const mockState = { stepsTaken: mock.mockStepsTaken };
      const mapped = mapStateToProps(mockState);
      expect(mapped.stepsTaken).toEqual(expected);
    });

    it('should add restingHeartRate data to props', () => {
      const expected = mock.restingHeartRate;
      const mockState = { restingHeart: mock.restingHeartRate };
      const mapped = mapStateToProps(mockState);
      expect(mapped.restingHeart).toEqual(expected);
    });
  });

  
});

