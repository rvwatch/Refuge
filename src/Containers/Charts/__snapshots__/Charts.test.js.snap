// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charts should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    heartRate={
        Array [
            Object {
              "time": "00:00:00",
              "value": 57,
            },
            Object {
              "time": "00:01:00",
              "value": 59,
            },
            Object {
              "time": "00:02:00",
              "value": 58,
            },
            Object {
              "time": "00:03:00",
              "value": 59,
            },
            Object {
              "time": "00:04:00",
              "value": 59,
            },
            Object {
              "time": "00:05:00",
              "value": 59,
            },
          ]
    }
    stepsTaken={
        Array [
            Object {
              "time": "00:00:00",
              "value": 0,
            },
            Object {
              "time": "00:01:00",
              "value": 0,
            },
            Object {
              "time": "00:02:00",
              "value": 0,
            },
            Object {
              "time": "00:03:00",
              "value": 0,
            },
            Object {
              "time": "00:04:00",
              "value": 0,
            },
            Object {
              "time": "00:05:00",
              "value": 0,
            },
            Object {
              "time": "00:06:00",
              "value": 0,
            },
            Object {
              "time": "00:07:00",
              "value": 0,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "render": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "render": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
